packages/components/src/dropdown-menu/types.ts

```diff
@@ -0,0 +1,146 @@
+	 */
+	label: string;
+	/**
+	 * A class name to apply to the dropdown menu's toggle element wrapper.
+	 */
+	className?: string;
+	/**
+	 * Properties of `popoverProps` object will be passed as props to the nested
+	 * `Popover` component.
+	 * Use this object to modify props available for the `Popover` component that
+	 * are not already exposed in the `DropdownMenu` component, e.g.: the
+	 * direction in which the popover should open relative to its parent node
+	 * set with `position` prop.
+	 */
+	popoverProps?: DropdownProps[ 'popoverProps' ];
+	/**
+	 * Properties of `toggleProps` object will be passed as props to the nested
+	 * `Button` component in the `renderToggle` implementation of the `Dropdown`
+	 * component used internally.
+	 * Use this object to modify props available for the `Button` component that
+	 * are not already exposed in the `DropdownMenu` component, e.g.: the tooltip
+	 * text displayed on hover set with `tooltip` prop.
+	 */
+	toggleProps?: ToggleProps;
+	/**
+	 * Properties of `menuProps` object will be passed as props to the nested
+	 * `NavigableMenu` component in the `renderContent` implementation of the
+	 * `Dropdown` component used internally.
+	 * Use this object to modify props available for the `NavigableMenu`
+	 * component that are not already exposed in the `DropdownMenu` component,
+	 * e.g.: the orientation of the menu set with `orientation` prop.
+	 */
+	menuProps?: Omit< Partial< NavigableMenuProps >, 'children' >;
+	/**
+	 * In some contexts, the arrow down key used to open the dropdown menu might
+	 * need to be disabled—for example when that key is used to perform another
+	 * action.
+	 *
+	 * @default false
+	 */
+	disableOpenOnArrowDown?: boolean;
+	/**
+	 * Text to display on the nested `Button` component in the `renderToggle`
+	 * implementation of the `Dropdown` component used internally.
+	 */
+	text?: string;
+	/**
+	 * Whether or not `no-icons` should be added to the menu's `className`.
+	 */
+	noIcons?: boolean;
+	/**
+	 * A [function render prop](https://reactjs.org/docs/render-props.html#using-props-other-than-render)
+	 * which should return an element or elements valid for use in a DropdownMenu:
+	 * `MenuItem`, `MenuItemsChoice`, or `MenuGroup`. Its first argument is a
+	 * props object including the same values as given to a `Dropdown`'s
+	 * `renderContent` (`isOpen`, `onToggle`, `onClose`).
+	 *
+	 * A valid DropdownMenu must specify a `controls` or `children` prop, or both.
+	 */
+	children?: ( callbackProps: DropdownCallbackProps ) => ReactNode;
+	/**
+	 * An array of objects describing the options to be shown in the expanded
+	 * menu. Each object should include an `icon` Dashicon slug string, a
+	 * human-readable `title` string, `isDisabled` boolean flag, and an `onClick`
+	 * function callback to invoke when the option is selected.
+	 *
+	 * A valid DropdownMenu must specify a `controls` or `children` prop, or both.
+	 */
+	controls?: Controls | NormalizedControls;
```

Comment:
I think that the `Controls` and `NormalizedControls` type names can be a bit confusing and hide the actual shape of the object. I'd probably remove those types and just use `DropdownOption`

<details>
<summary>Something like this</summary>

```diff
diff --git a/packages/components/src/dropdown-menu/index.tsx b/packages/components/src/dropdown-menu/index.tsx
index c21cfa456e..805bcd0661 100644
--- a/packages/components/src/dropdown-menu/index.tsx
+++ b/packages/components/src/dropdown-menu/index.tsx
@@ -14,7 +14,7 @@ import { menu } from '@wordpress/icons';
 import Button from '../button';
 import Dropdown from '../dropdown';
 import { NavigableMenu } from '../navigable-container';
-import type { Controls, DropdownMenuProps, NormalizedControls } from './types';
+import type { DropdownMenuProps, DropdownOption } from './types';
 
 function mergeProps<
 	T extends { className?: string; [ key: string ]: unknown }
@@ -139,7 +139,7 @@ function DropdownMenu( dropdownMenuProps: DropdownMenuProps ) {
 	}
 
 	// Normalize controls to nested array of objects (sets of controls)
-	let controlSets: NormalizedControls;
+	let controlSets: DropdownOption[][];
 	if ( controls?.length ) {
 		// @ts-expect-error The check below is needed because `DropdownMenus`
 		// rendered by `ToolBarGroup` receive controls as a nested array.
@@ -147,7 +147,7 @@ function DropdownMenu( dropdownMenuProps: DropdownMenuProps ) {
 		if ( ! Array.isArray( controlSets[ 0 ] ) ) {
 			// This is not ideal, but at this point we know that `controls` is
 			// not a nested array, even if TypeScript doesn't.
-			controlSets = [ controls as Controls ];
+			controlSets = [ controls as DropdownOption[] ];
 		}
 	}
 
diff --git a/packages/components/src/dropdown-menu/types.ts b/packages/components/src/dropdown-menu/types.ts
index a52e04ff08..6e4a838e00 100644
--- a/packages/components/src/dropdown-menu/types.ts
+++ b/packages/components/src/dropdown-menu/types.ts
@@ -11,7 +11,7 @@ import type { DropdownProps } from '../dropdown/types';
 import type { Props as IconProps } from '../icon';
 import type { NavigableMenuProps } from '../navigable-container/types';
 
-type DropdownOption = {
+export type DropdownOption = {
 	/**
 	 * The Dashicon icon slug to be shown for the option.
 	 */
@@ -44,9 +44,6 @@ type DropdownOption = {
 	role?: HTMLElement[ 'role' ];
 };
 
-export type Controls = DropdownOption[];
-export type NormalizedControls = DropdownOption[][];
-
 type DropdownCallbackProps = {
 	isOpen: boolean;
 	onToggle: () => void;
@@ -142,5 +139,5 @@ export type DropdownMenuProps = {
 	 *
 	 * A valid DropdownMenu must specify a `controls` or `children` prop, or both.
 	 */
-	controls?: Controls | NormalizedControls;
+	controls?: DropdownOption[] | DropdownOption[][];
 };
```
</summary>

Reply:
Works for me! ✅ b38f617df8